doc/sphinx/source/developer_guide/logging.rst

```diff
@@ -0,0 +1,264 @@
+   using ``telnet 192.168.42.1``.
+3. Navigate to the Paparazzi folder using ``cd data/ftp/internal_000/paparazzi``.
+4. You need to restart the autopilot for the messages to show up. Run ``killall -9 ap.elf``, then
+   ``./ap.elf`` to restart the autopilot.
+5. When you are finished and the drone is landed, kill the autopilot by pressing Ctrl+C twice, then
+   disconnect using Ctrl+D.
+
+.. note::
+
+  The target address of the ``telnet`` command may be different than what is reported here. You can
+  double check by going into the Makefile of your drone and looking for the ``HOST`` variable. For
+  example, the Makefile for the Bebop drone can be found in ``conf/boards/bebop.makefile``
+
+.. warning::
+
+  It is very easy to accidentally start multiple autopilot processes when connected to the drone by
+  WiFi. This can cause flickering in the GCS as it receives duplicate messages, as well as slowing down
+  the execution time. You can terminate all autopilot processes simultaneously by running ``killall -9 ap.elf``
+
+File Logger
+^^^^^^^^^^^^^^
+
+Logging data onboard on a Linux based OS is pretty straightforward. The Bebop drone, for example, is
+equipped with onboard storage, which can be used to store the logs. First, it is necessary to add the
+``logger_file`` module in your airframe and specifying the path to the internal storage.
+
+.. code-block:: xml
+
+  <firmware name="rotorcraft"/>
+    <module name="logger_file">
+      <define name="FILE_LOGGER_PATH" value="/data/ftp/internal_000"/>
+    </module>
+  </firmware>
+
+Then, in the C file ``sw/airborne/modules/loggers/file_logger.c`` add the header of the module that
+contains the variables that you want to log. Within ``file_logger.c`` you can specify which variables
+you want to have logged in the same fashion as normal ``fprintf`` as described in the previous sesction.
+The variable names and types have to be specified.
+
+The file logger needs to be manually started and stopped by going to the Paparazzi GCS. In the module
+settings tab navigate to Settings -> Modules. Here you can start and stop a log. To retrieve the log file
+connect to the drone by WiFi and using either a browser or an FTP client (like the Linux standard Files application)
+connect to ``ftp://196.168.42.1``. This will take you to the FTP folder of the Bebop. Browse to ``/Internal000/``.
+Here we can find the .csv files, named ``00000.csv`` to ``0000X.csv`` depending on the flights you logged.
+Copy the files to your computer for analysis.
+
+
+Logging on ChibiOS based OS
+-----------------------------
+
+Logging on ChibiOS systems can be achieved using the ``flight_recorder`` module, which depends on a
+few other modules that also need to be included in your airframe.
+
+.. code-block:: xml
+
+  <firmware name="rotorcraft">
+    <module name="tlsf"/>
+    <module name="pprzlog"/>
+    <module name="logger" type="sd_chibios"/>
+    <module name="flight_recorder"/>
```

Comment:
With the new dependency solver, it is only necessary to load `flight_recorder` module.

Also, when using `sd_chibios` only, a log file is opened and can be used to log any type of data in ASCII (or even binary if needed) like a regular file. When loading the flight recorder module, two separated files are opened. Have a look at the `meteo_stick` module for an example of using the different options depending of the module's configuration.